app/controllers/v0/sessions_controller.rb

```diff
@@ -37,7 +37,13 @@ def saml_callback
         obscure_token = Session.obscure_token(@session.token)
         Rails.logger.info("Logged in user with id #{@session.uuid}, token #{obscure_token}")
       else
-        log_errors
+        message = <<-MESSAGE.strip_heredoc
```

Comment:
and if you want to get super nit picky you could decompose the whole conditional. `@saml_response.is_valid?` becomes a guard clause inside `persist_session_and_user` and the code within the if above (lines 34-38) is also extracted to a method.